herddb-core/src/main/java/herddb/core/TableManager.java

```diff
@@ -2172,15 +2220,14 @@ private TableCheckpoint checkpoint(double dirtyThreshold, double fillThreshold,
             for (DataPage dataPage : newPages.values()) {
                 if (!dataPage.isEmpty()) {
                     bufferPageSize -= flushNewPageForCheckpoint(dataPage, buffer);
-//                    dataPage.makeImmutable();
                     ++flushedNewPages;
                     flushedRecords += dataPage.size();
                 }
             }
 
             /* Flush remaining records */
             if (!buffer.isEmpty()) {
-                createImmutablePage(buffer, bufferPageSize);
+                createImmutablePage(buffer, bufferPageSize, /* best guess */ keepFlushedPageInMemory);
```

Comment:
can you explain better "best guess" ?
tomorrow we won't remebber about this patch, it is better to add a comment here

Reply:
I'll reword better. It's just a fast note written 1 min before commit :(

This is a "best guess" because we don't known if records that needed to be kept in memory were already be flushed during newPage filling (line 2222). So we still use keepFlushedPageInMemory (possibily true) even if remaining records came from an old unused page.

Reply:
Added a more explaining comment